--- src/game.rs
+++ src/game.rs
@@ -199,6 +199,13 @@
     let (tx, mut _rx) = broadcast::channel::<String>(100);
     let (mut user_ws_tx, mut user_ws_rx) = ws.split();
     let mut die = false;
+    {
+        let gs = game_state.lock().await;
+        if gs.banned_players.contains(&login_name) {
+            let _ = user_ws_tx.send(Message::text(serde_json::to_string(&packets::Outgoing::Banned {}).unwrap())).await;
+            return;
+        }
+    }
     {
         let mut gs = game_state.lock().await;
         gs.sendable.set_host(&login_name);
